### PR TITLE
Added event info to event protocol

### PR DESCRIPTION
**Pull request by raguiar2**
I've been using this calendarkit library for a while, and I noticed that there was nothing allowing users to actually store info about the event inside of the event descriptor protocol, so I added it. 

This change could be useful for being able to get information about the event when the user calls dayViewDidSelectEventView, for example.

### CODE REVIEW

**Comment by richardtop** [comment]
Hi @raguiar2 thanks for the pull request, however `userInfo` serves exactly that purpose. Also, if needed, its recommended that you create your custom `Event` class.

**Comment by richardtop** [comment]
How is your addition is different from `public var userInfo: Any?` ?

**Comment by raguiar2** [comment]
I don’t think userInfo:Any? Is exposed through the event descriptor. That means you can’t use it in the functions called when you tap or long press the event. If you want, I can edit the code to expose it through the delegate instead of adding a new field.

**Comment by raguiar2** [comment]
I also don’t think adding a custom event class will work because you can’t access it through the EventDescriptor protocol anyway

**Comment by richardtop** [comment]
@raguiar2 `EventDescriptor` protocol is used only to communicate the absolute necessities about event with the `CalendarKit`. Therefore, the `userInfo` hasn't been included in the protocol intentionally. The `Event` class is a default implementation conforming to `EventDescriptor` which should fit the majority of the use cases.

However, if you'd like to add some specific domain logic or alter e.g. color calculation, you'd be better of with creating your own `CustomEventClass` conforming to `EventDescriptor`.

The problem you experience is that you'll need to cast `EventDesccriptor` to your particular class, for example like this:

```Swift
  // MARK: DayViewDelegate
  override func dayViewDidSelectEventView(_ eventView: EventView) {
    guard let descriptor = eventView.descriptor as? Event else {
      return
    }
    print("Event has been selected: \(descriptor) \(String(describing: descriptor.userInfo))")
  }

```

This is the sample code from the example project, but you could use your own custom class instead.



In short, 
**EventDescriptor contains only the info needed for CalendarKit to display your event. `userInfo` is by definition not used by CalendarKit, so its implementation is up to a particular class (you may create your own event class and add `eventInfo` instead of `userInfo` if that fits you better).**

**Comment by raguiar2** [comment]
Ah, I see. Thanks for clarifying! I'll close this.
